### PR TITLE
Remove escape characters in observability data

### LINKED ISSUE #37830: [Bug]: Ballerina resource path escape characters present in Observability data
### Description

Ballerina escape characters are simply used at source code level when the user requires using a special character in their code. When the user wants to use `-` in their resource function, they can currently use `\-` to escape and use it.

```ballerina
import ballerina/http;

service / on new http:Listener(9090) {
    resource function get greeting\-test() returns string {
        return "Hello";
    }
}
```

However, these escape character (`\`) should not be present in the Observability data. However, it was observed that they are present. We need to remove the escape characters in the Observability data.

Eg:- User writing the above code should result in the resource path to be recorded as `greeting-test`

### Steps to Reproduce

1. Use the following code (has additional logic to filter and check a single metric for the resource path in question, using a in memory metrics store extension used for testing purposes).
   ```ballerina
   import ballerina/http;
   import ballerina/observe.mockextension;
   
   service / on new http:Listener(9090) {
       resource function get greeting\-test() returns string {
           return "Hello";
       }
   
       resource function get requestsTotalMetrics() returns mockextension:Counter[] {
           mockextension:Counter[] filteredMetrics = [];
           foreach mockextension:Counter c in mockextension:getMetrics().counters {
               if c.id.name == "requests_total" {
                   foreach mockextension:Tag t in c.id.tags {
                       if t.key == "src.resource.path" && t.value.startsWith("/greeting") {
                           filteredMetrics.push(c);
                           break;
                       }
                   }
               }
           }
           return filteredMetrics;
       }
   }
   ```

2. Use the following `Config.toml` entry to enable the extension
   ```toml
   [ballerina.observe]
   metricsEnabled=true
   metricsReporter="mock"
   ```

3. Run the ballerina code as a project using `bal run`.

4. Call the resource path in question.
    ```bash
    curl http://localhost:9090/greeting-test
    ```

5. Call the additional endpoint for viewing the collected metrics.
    ```bash
    curl http://localhost:9090/requestsTotalMetrics | jq
    ```

6. Note the tags which are added by the Observability runtime and how the values are containing `\`s.
    Sample output (Note `src.resource.path` and `entrypoint.function.name` tags) -
    ```
    [
      {
        "id": {
          "name": "requests_total",
          "description": "Total number of requests",
          "tags": [
            {
              "key": "listener.name",
              "value": "http",
              "hashCode": -10011292
            },
            {
              "key": "src.resource.path",
              "value": "/greeting\\-test",
              "hashCode": -1043146127
            },
            {
              "key": "http.status_code_group",
              "value": "2xx",
              "hashCode": 475734207
            },
            {
              "key": "src.object.name",
              "value": "/",
              "hashCode": 11132704
            },
            {
              "key": "protocol",
              "value": "http",
              "hashCode": -596094799
            },
            {
              "key": "entrypoint.function.module",
              "value": "nadundesilva/temp:0.1.0",
              "hashCode": 1983732311
            },
            {
              "key": "http.url",
              "value": "/greeting-test",
              "hashCode": 981751727
            },
            {
              "key": "entrypoint.resource.accessor",
              "value": "get",
              "hashCode": 802079584
            },
            {
              "key": "entrypoint.function.name",
              "value": "/greeting\\-test",
              "hashCode": -95045797
            },
            {
              "key": "src.service.resource",
              "value": "true",
              "hashCode": -7446146
            },
            {
              "key": "src.resource.accessor",
              "value": "get",
              "hashCode": -925933542
            },
            {
              "key": "entrypoint.service.name",
              "value": "/",
              "hashCode": 619115708
            },
            {
              "key": "src.module",
              "value": "nadundesilva/temp:0.1.0",
              "hashCode": 852242643
            },
            {
              "key": "src.position",
              "value": "service.bal:5:5",
              "hashCode": 1292353336
            },
            {
              "key": "http.method",
              "value": "GET",
              "hashCode": 175076048
            }
          ],
          "hashCode": -1149741011
        },
        "value": 1
      }
    ]
    ```

### Affected Version(s)

```
Ballerina 2201.1.2 (Swan Lake Update 1)
Language specification 2022R2
Update Tool 1.3.10
```
And possibly other versions

### OS, DB, other environment details and versions

Any

### Related area

Obsevability

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title
Fixes #37830

Tests will be added in https://github.com/ballerina-platform/module-ballerinai-observe/pull/48

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java [hunk 1]
```diff
@@ -183,9 +183,8 @@ void instrumentPackage(BIRPackage pkg) {
                 }
                 if (serviceName == null) {
                     String basePath = this.svcAttachPoints.get(typeDef.name);
-                    serviceName = Objects.requireNonNullElseGet(basePath, () ->
-                            pkg.packageID.orgName.value + "_" + pkg.packageID.name.value + "_svc_" +
-                            defaultServiceIndex++);
+                    serviceName = basePath == null ? pkg.packageID.orgName.value + "_" + pkg.packageID.name.value +
+                            "_svc_" + defaultServiceIndex++ : Utils.unescapeBallerina(basePath);
```
Comment (gabilang) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java:187]:
```suggestion
                    serviceName = basePath == null ? pkg.packageID.orgName.getValue() + "_" + pkg.packageID.name.getValue() +
                            "_svc_" + defaultServiceIndex++ : Utils.unescapeBallerina(basePath);
```
Shall we use the getter method to get the name strings? The field `value` in the class `Name.java` should be private (at least in future). A [TODO](https://github.com/ballerina-platform/ballerina-lang/blob/c6f9327976c7eb64dd2d93fa75a5b4d053c7cdfa/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/Name.java#L25) has already been added for that.

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Review by gabilang — DISMISSED** [review]
LGTM

**Review by HindujaB — APPROVED** [review]
LGTM

**Pull request merged**